### ISSUE TITLE
Why does StreamManager not support uploading files to S3 buckets in another accounts?

### ISSUE THREAD

**Issue by cgddrd-gardin** [aws-greengrass/aws-greengrass-stream-manager-sdk-python#12]
Hi,

I've noted in the documentation for StreamManager (https://docs.aws.amazon.com/greengrass/v2/developerguide/stream-export-configurations.html#export-to-s3) that uploads to S3 buckets in other accounts is not supported as per the below screenshot. My question is, what prevents us from setting up cross-account access to an S3 bucket in another account, using an IoT Core role alias?

![image](https://user-images.githubusercontent.com/86467305/190967908-3a2ec653-0c12-4c8e-83d9-fc8f0b12c51b.png)

**Comment by jbutler** [comment]
Hi Connor --

Greengrass uses temporary credentials to access S3. Behind the scenes, AWS IoT is calling the AWS STS service to assume role credentials and return those to the device.

Giving Greengrass access to an S3 bucket in a different account would require setting up a bucket policy that gives access to the assumed role which Greengrass uses.

Bucket policies giving access to role session principals must specify a role session name. In this case, the role session name is randomly generated. There is no way to know it in advance in order to be able to set up the bucket policy.

Refer to:
https://docs.aws.amazon.com/iot/latest/developerguide/authorizing-direct-aws.html
https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_elements_principal.html#principal-role-session

TL;DR there's no way to set up a bucket policy that gives Greengrass access since Greengrass uses temporary credentials with a random session name.

**Comment by cgddrd-gardin** [comment]
Hey Joe, 

That info is very insightful, thanks!
